### PR TITLE
feat: added rate limiting and updated dependencies

### PR DESCRIPTION
**Pull request by peterMangrove**

### CODE REVIEW

File: src/cmd/server.ts [hunk 1]
```diff
@@ -133,17 +139,41 @@ async function main() {
...
+      context: async ({ req, res }) => ({
+        prisma,
+      }),
+    })
+  );
+
+  httpServer.listen({ port: PORT });
+
+  // await new Promise((resolve) => httpServer.listen({ port: PORT }));
+  console.log(`🚀 Server ready at http://localhost:4000/`);
```
Comment (lnist) [inline src/cmd/server.ts]: remove commented out code? Get url from httpServer?

**Review by lnist — COMMENTED** [review]
Github complains about some unused vars, and one small comment from me :)

**Review by espendk — CHANGES_REQUESTED** [review]
We should make this controllable by env vars, so we can easily update the parameters in production

File: src/cmd/server.ts [hunk 2]
```diff
@@ -133,17 +139,41 @@ async function main() {
     validate: false,
   });
 
+  const app = express();
+
+  const limiter = rateLimit({
+    windowMs: 15 * 60 * 1000, // 15 minutes
+    max: 100, // Limit each IP to 100 requests per `window` (here, per 15 minutes)
```
Comment (espendk) [inline src/cmd/server.ts]: We should make this controllable by env vars, so we can easily update the parameters in production

**Review by espendk — CHANGES_REQUESTED** [review]

File: src/cmd/server.ts (hunk 2 above)
Comment (peterMangrove) [reply src/cmd/server.ts]: Makes sense. Will make that change

Comment (peterMangrove) [reply src/cmd/server.ts]: Done

File: src/cmd/server.ts (hunk 1 above)
Comment (peterMangrove) [reply src/cmd/server.ts]: Done

**Pull request merged**